### DISCUSSION TITLE
Possible Improvement in Field Validation

### DISCUSSION THREAD

**Discussion by thiagao0860** [O365/python-o365#962] [General]
Hi, thanks in advance for this lib it helps me a lot with my activities. but I think there is a problem in method responsible to check field validation before update it: 

```
 def _valid_field(self, field):
        # Verify the used field names are valid internal field names
        valid_field_names = self.fields if self.fields \
            else self._parent.column_name_cw.values() \
            if self._parent \
            else None
        if valid_field_names:
            return field in valid_field_names

        # If no parent is given, and no internal fields are defined assume correct, API will check
        return True
```

As the validation of parent fields is done after an "else" statement it will only check for this if there are no child fields, but this is a problem because if you have a null field on child, it yet can be a valid in your parent. I made this change on a fork and it works well for me:

````
def _valid_field(self, field):
        if field is self.fields:
            return True
        elif field in self._parent.column_name_cw.values():
            return True
        else:
            return False
       
        ...
````

So I don't know if you have some special assumption about this and this is why I didn't make a pull, but I think it's good warn

**Comment by alejcas** [comment]
Your code have some bugs. For example. self._parent can be None. In this case it will produce an error.

I didn't write this portion of the library code, but pycharm tells me that @jtweeder did it a long time ago.

Maybe he can help here?

Thanks

**Reply by thiagao0860** [reply]
Yep, my point it's not include this code directly (it's an example) but show that the current code is ignoring the case that parent has fields but child doesn't.

**Comment by jtweeder** [comment]
I should have some time in the coming days to refresh my memory and provide some input. 

I believe the intent of that segment was to check known field names against what sharepoint was expecting based on the known field names.  Giving it an opportunity to fail before wasting an API call for sending the wrong column name.
